hazelcast/src/main/java/com/hazelcast/instance/DefaultNodeExtension.java

```diff
@@ -250,4 +297,13 @@ public boolean triggerForceStart() {
     public String createMemberUuid(Address address) {
         return UuidUtil.createMemberUuid(address);
     }
+
+    // obtain cluster version, if already initialized, otherwise this node's version
+    private Version getClusterOrNodeVersion() {
+        if (node.getClusterService() != null && node.getClusterService().getClusterVersion() != null) {
```

Comment:
This is only used when delivering current cluster version to a freshly registered listener with the first callback to `onClusterVersionChange`.

In specific case of serialization service, on node startup cluster version will be `null`, so I am not sure how you would like to handle this on SS side; my original thought was that SS listener shouldn't have to deal with `null` cluster version but feel free to propose any improvement.

I guess other cluster version listeners are only registered after node has joined, this is only relevant to cluster version listeners registered before node join (so currently this is only SS).